lib/http.js

```diff
@@ -31,12 +31,46 @@ exports.request = function(options, cb) {
   return new ClientRequest(options, cb);
 };
 
+exports.requestAsync = function(options) {
+  var resolveRequest = null;
+  const getResponse = new Promise((resolve) => {
+    resolveRequest = resolve;
+  });
+  const request = exports.request(options, (res) => {
+    resolveRequest(res);
+  });
+  return {
+    request,
+    response: getResponse
+  };
+};
+
 exports.get = function(options, cb) {
   var req = exports.request(options, cb);
   req.end();
   return req;
 };
 
+exports.getAsync = function(options) {
+  var resolveRequest = null;
+  var rejectRequest = null;
+  const getResponse = new Promise((resolve, reject) => {
+    resolveRequest = resolve;
+    rejectRequest = reject;
+  });
+  const request = exports.request(options, (res) => {
+    resolveRequest(res);
+  });
+  request.on('error', rejectRequest);
+  getResponse.then(() => {
```

Comment:
> i just dont get the problem

1. There is a `maxListeners` limit for the `EventEmitter`'s that is 10 by default. If more than `maxListeners` handlers assigned then `EventEmitter` puts warning message into stderr. Here we assign additional listener internally. Someone, may rely on `maxListeners` limit, so they may get warning earlier than it's expected (on 9th listener).

2. The reason `maxListeners` was introduced is to avoid memory leaks with dangling event handlers. Here we remove error handler if promise is successfully resolved, but we don't do that on rejection. The easiest way to avoid it is to use: `.once()` instead of `.on()`